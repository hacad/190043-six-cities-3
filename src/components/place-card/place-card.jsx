import React from "react";
import PropTypes from "prop-types";
import PlacePropType from "../prop-types/place.js";

const PlaceCard = (props) => {
  const {place, onClickHeader, onActivate, onDeactivate} = props;
  const {type, img, category, name, price, rating} = place;

  return (
    <article className="cities__place-card place-card"
      onMouseEnter={() => {
        onActivate(place);
      }}
      onMouseLeave={() => {
        onDeactivate(place);
      }}
    >
      {category
        ? (
          <div className="place-card__mark">
            <span>{category}</span>
          </div>
        )
        : ``
      }
      <div className="cities__image-wrapper place-card__image-wrapper">
        <a href="#">
          <img className="place-card__image" src={img} width="260" height="200" alt="Place image" />
        </a>
      </div>
      <div className="place-card__info">
        <div className="place-card__price-wrapper">
          <div className="place-card__price">
            <b className="place-card__price-value">{price.currency}{price.value}</b>
            <span className="place-card__price-text">&#47;&nbsp;night</span>
          </div>
          <button className="place-card__bookmark-button button" type="button">
            <svg className="place-card__bookmark-icon" width="18" height="19">
              <use xlinkHref="#icon-bookmark"></use>
            </svg>
            <span className="visually-hidden">To bookmarks</span>
          </button>
        </div>
        <div className="place-card__rating rating">
          <div className="place-card__stars rating__stars">
            <span style={{width: `${rating}%`}}></span>
            <span className="visually-hidden">Rating</span>
          </div>
        </div>
        <h2 className="place-card__name" onClick={onClickHeader}>
          <a href="#">{name}</a>
        </h2>
        <p className="place-card__type">{type}</p>
      </div>
    </article>
  );
};

PlaceCard.propTypes = {
  place: PlacePropType.isRequired,
  onClickHeader: PropTypes.func.isRequired,
  onActivate: PropTypes.func.isRequired,
  onDeactivate: PropTypes.func.isRequired
};

export default PlaceCard;
