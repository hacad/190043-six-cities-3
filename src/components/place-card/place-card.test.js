import React from "react";
import renderer from "react-test-renderer";
import {PlaceCard} from "./place-card.jsx";

/* eslint-disable camelcase */

const testPlace = {
  id: 1,
  city: {
    name: `Amsterdam`,
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10
    }
  },
  type: `apartment`,
  previewImage: `img/apartment-01.jpg`,
  isPremium: false,
  title: `Beautiful &amp; luxurious apartment at great location`,
  price: 120,
  rating: 93,
  isFavorite: false,
  location: {
    latitude: 52.370216,
    longitude: 4.895168
  }
};

it(`PlaceCard correctly renders after relaunch`, () => {
  const placeCard = renderer
    .create(<PlaceCard
      place={testPlace}
      onClickHeader={jest.fn()}
      onActivate={jest.fn()}
      onDeactivate={jest.fn()}
      toggleFavorite={jest.fn()}/>)
    .toJSON();

  expect(placeCard).toMatchSnapshot();
});

/* eslint-enable */
