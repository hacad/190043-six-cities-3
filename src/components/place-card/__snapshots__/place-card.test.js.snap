// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCard correctly renders after relaunch 1`] = `
<article
  className="cities__place-card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  
  <div
    className="cities__image-wrapper__image-wrapper place-card__image-wrapper"
  >
    <a
      href="/offer/1"
      onClick={[Function]}
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="img/apartment-01.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className=" place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          120
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <<div />
        classNamePrefix="place-card"
        iconHeight={19}
        iconWidth={18}
        isFavorite={false}
        offerId={1}
      />
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "undefined%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
      onClick={[MockFunction]}
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        Beautiful &amp; luxurious apartment at great location
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      apartment
    </p>
  </div>
</article>
`;
